h/views/status.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import logging
+
+from h.util.view import json_view
+from h.exceptions import APIError
+
+
+log = logging.getLogger(__name__)
+
+
+@json_view(route_name='status')
+def status(request):
+    _check_database(request)
+    _check_search(request)
+    _check_realtime(request)
+    return {'status': 'okay'}
+
+
+def _check_database(request):
+    try:
+        request.db.execute('SELECT 1')
+    except Exception as exc:
+        log.exception(exc)
+        raise APIError('Database connection failed')
+
+
+def _check_search(request):
+    try:
+        info = request.es.conn.cluster.health()
+        if info['status'] not in ('green', 'yellow'):
+            raise APIError('Search cluster state is %s' % info['status'])
+    except Exception as exc:
+        log.exception(exc)
+        raise APIError('Search connection failed')
+
+
+def _check_realtime(request):
+    try:
+        request.realtime.connection.release()
+        request.realtime.connection.connect()
```

Comment:
As discussed in Slack, I had some reservations about what happens to `ping` requests when the workers are already busy because of a long job queue, but it turns out that `ping` control requests are not enqueued as jobs, so we don't have to worry about this. This is done now 👍